kythe/cxx/indexer/cxx/clang_utils.cc

```diff
@@ -45,12 +45,48 @@ clang::SourceLocation GetLocForEndOfToken(
                                            source_manager, lang_options);
 }
 
-// TODO(zarko): Update this to handle member specializations.
-const clang::Decl* FindSpecializedTemplate(const clang::Decl* decl) {
+const clang::Decl* DereferenceMemberTemplates(const clang::Decl* decl,
+                                              bool use_mts) {
+  if (!use_mts) return decl;
+  if (const auto* ft = clang::dyn_cast<clang::FunctionTemplateDecl>(decl)) {
```

Comment:
Use a clang::ConstDeclVisitor for this rather than the else if chains? It will be (marginally) faster, but generally keep the code better structured.

Reply:
Done.